shimming-toolbox/shimmingtoolbox/files/file.py

```diff
@@ -0,0 +1,169 @@
+from __future__ import annotations
+import logging
+import nibabel as nib
+import os
+import numpy as np
+from functools import wraps
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+def safe_getter(default_value=None):
+    """Decorator that catches errors in getter functions and returns a default value."""
+    def decorator(func):
+        @wraps(func)
+        def wrapper(self, *args, **kwargs):
+            try:
+                return func(self, *args, **kwargs)
+            except Exception as e:
+                logger.error(f"Error in {func.__name__}: {e}")
+                # terminate the program if the error is critical
+                if isinstance(e, (NameError, ValueError)):
+                    raise e
+                return default_value
+        return wrapper
+    return decorator
+
+NIFTI_EXTENSIONS = ('.nii.gz', '.nii')
+DEFAULT_SUFFIX = '_saved.nii.gz'
+
+class NiftiFile:
+    def __init__(self, path_nii: str) -> None:
```

Comment:
```suggestion
    def __init__(self, fname_nii: str) -> None:
```
See our [naming conventions](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Programming%3A-Naming-conventions-(variables,-files)) (based on SCT)